Sources/Vehicle/Templates/EntitiesGridTemplate.swift

```diff
@@ -0,0 +1,122 @@
+import CarPlay
+import Foundation
+import HAKit
+import PromiseKit
+import Shared
+
+@available(iOS 16.0, *)
+class EntitiesGridTemplate {
+    
+    private let entityIconSize: CGSize = CGSize(width: 64, height: 64)
+    private var stateSubscriptionToken: HACancellable?
+    private let title: String
+    private let domain: String
+    private var server: Server
+    private var entities: [HAEntity] = []
+    private var gridTemplate: CPGridTemplate?
+    private var gridPage: Int = 0
+    
+    enum GridPage {
+        case Next
+        case Previous
+    }
+    
+    init(title: String, domain: String, server: Server, entities: [HAEntity]) {
+        self.title = title
+        self.domain = domain
+        self.server = server
+        self.entities = entities
+    }
+    
+    public func getTemplate() -> CPGridTemplate {
+        guard let gridTemplate = gridTemplate else {
+            gridTemplate = CPGridTemplate(title: title, gridButtons: getGridButtons())
+            return gridTemplate!
+        }
+        return gridTemplate
+    }
+    
+    func getGridButtons() -> [CPGridButton] {
+        var items: [CPGridButton] = []
+
+        let entitiesSorted = entities.sorted(by: { $0.attributes.friendlyName ?? "" < $1.attributes.friendlyName ?? "" })
+        
+        let entitiesPage = entitiesSorted[(gridPage * CPGridTemplateMaximumItems) ..< min((gridPage * CPGridTemplateMaximumItems) + CPGridTemplateMaximumItems, entitiesSorted.count)]
+        
+        for entity in entitiesPage {
+            let item = CPGridButton(titleVariants: ["\(entity.attributes.friendlyName!) - \(entity.getFriendlyState())"], image: entity.getIcon() ?? MaterialDesignIcons.bookmarkIcon.image(ofSize: entityIconSize, color: nil), handler: { button in
```

Comment:
This throws an error in my testing:
```suggestion
            let item = CPGridButton(titleVariants: ["\(entity.attributes.friendlyName ?? "<Unnamed entity>") - \(entity.getFriendlyState())"], image: entity.getIcon() ?? MaterialDesignIcons.bookmarkIcon.image(ofSize: entityIconSize, color: nil), handler: { button in
```